src/printer.js

```diff
@@ -1442,8 +1442,11 @@ function genericPrintNoParens(path, options, print) {
       ]);
     case "DeclareInterface":
     case "InterfaceDeclaration": {
-      const parent = path.getParentNode(1);
-      if (parent && parent.type === "DeclareModule") {
+      const nodeSource = options.originalText.slice(0, util.locStart(n));
+      if (
+        n.type === "DeclareInterface" ||
+          (options.parser === "flow" && nodeSource.match(/declare\s*$/))
```

Comment:
Do you want to compute the regex on the fly (with `new RegExp`) too so that the matching word could be anything else than `declare`?